pr_agent/git_providers/github_provider.py

```diff
@@ -28,7 +29,14 @@ def __init__(self, pr_url: Optional[str] = None):
             self.installation_id = None
         self.max_comment_chars = 65000
         self.base_url = get_settings().get("GITHUB.BASE_URL", "https://api.github.com").rstrip("/")
-        self.base_url_html = self.base_url.split("api/")[0].rstrip("/") if "api/" in self.base_url else "https://github.com"
+        if self.base_url.endswith("/api/v3"):
+            # enterprise server with api/v3
+            self.base_url_html = self.base_url[:len("/api/v3")]
+        elif self.base_url == "https://api.github.com":
+            # github cloud
+            self.base_url_html = "https://github.com"
+        else:
+            self.base_url_html = self.base_url
```

Comment:
explain this change.
why is it needed ?

Give concrete examples - what didn't work before, and is working now due to this change ?